### PR TITLE
Protecting against malleability/replay attacks

### PR DESCRIPTION
**Pull request by seresistvanandras**
Malleability attacks can be devastating for TLPs and their applications. This PR is an easy fix to prevent replaying valid ballots from other users by incorporating the `msg.sender` in the `paramtersHash` variable, whenever a Fiat-Shamir challenge is computed. I think, this might not be necessary for the Wesolowski proof, but I also attached it there. Maybe there replayability is rather a feature and not a bug, since we want to allow a single party to compute the VDF proof but to allow anyone to submit that to the chain. This might not be the case if we want to give rewards to incentivize the Wesolowski VDF proof submissions.

### CODE REVIEW

File: src/CicadaVote.sol [hunk 1]
```diff
@@ -191,6 +192,7 @@ abstract contract CicadaVote {
         if (parametersHash != vote.parametersHash) {
             revert ParametersHashMismatch();
         }
+        parametersHash = keccak256(abi.encode(pp, msg.sender));
```
Comment (moodlezoup) [inline src/CicadaVote.sol:195]: we don't need this line

File: src/CicadaVote.sol [hunk 2]
```diff
@@ -321,7 +323,7 @@ abstract contract CicadaVote {
         internal
         view
     {
-        bytes32 parametersHash = keccak256(abi.encode(pp));
+        bytes32 parametersHash = keccak256(abi.encode(pp, msg.sender));
```
Comment (moodlezoup) [inline src/CicadaVote.sol:326]: I think we can leave this one as is, since this function is only used to finalize a vote.

**Review by moodlezoup — CHANGES_REQUESTED** [review]
Good catch, but I think we only need to update the `parameterHash` passed to `_verifyBallotValidity` (in `_castBallot`)

File: src/CicadaVote.sol (hunk 1 above)
Comment (seresistvanandras) [reply src/CicadaVote.sol]: Fixed.

File: src/CicadaVote.sol (hunk 2 above)
Comment (seresistvanandras) [reply src/CicadaVote.sol]: Yeah, I agree. Fixed. Everyone should be able to finalize to vote, once the voting period has ended.

**Review by moodlezoup — APPROVED** [review]
thanks!  😎

**Comment by moodlezoup** [comment]
oh wait I just realized this breaks the end-to-end test :P
https://github.com/a16z/cicada/blob/main/test/CicadaVote.t.sol#L55
Let's fix that before merging

**Comment by seresistvanandras** [comment]
Yeah, indeed! I will fix the tests soon!

**Pull request merged**